### DISCUSSION TITLE
Terminated repos

### DISCUSSION THREAD

**Discussion by BellezaEmporium** [iptv-org/community#332] [Show and tell]
GH has decided to disable both iptv and epg services for violation of their Terms & Conditions. While I am not on the rights to do anything against it, I guess it's time we say RIP.

**Comment by themak1985** [comment]
Should we transition to a self hosted gitlab?

**Reply by ferris007** [reply]
Absolutely!

**Comment by Daniel15** [comment]
The email from GitHub says "You may contact GitHub Support for more information or to appeal this decision: [Contact GitHub Support](https://support.github.com/contact/reinstatement)". Has anyone attempted this?

They said they removed it based on their Private Information Removal Policy, but I'm not sure how this repo would violate that policy. We should get clarification from them. https://docs.github.com/en/site-policy/content-removal-policies/github-private-information-removal-policy

Edit: I opened a ticket asking for clarification. Let's see what they say.

**Comment by BellezaEmporium** [comment]
It may be that only admins have recieved it. I have no mails from GitHub saying anything on my side.

EDIT : In my opinion, someone tried to DMCA them, but wanted to remove the repo ASAP. They might have used the " increased likelihood of unauthorized access to physical or network facilities." as a result of that takedown.

Though the only " increased likelihood of unauthorized access to physical or network facilities." reason, being litterally showing links that are available in the Internet is what they consider a violation of their policy.

**Comment by what-a** [comment]
GitHub objecting to the EPG is mind boggling.   
😞

**Comment by BellezaEmporium** [comment]
Someone named Matthew Gall has made a mirror of iptv-org/iptv, available here : https://matthewgall.codes/mirrors/iptv.

Last update is 3 days ago.

**Comment by Dum4G** [comment]
It's not the first time for the repository to be DMCAed. We usually had those cases resolved by contacting GH support and taking down inappropriate links. Stay tuned

**Comment by BellezaEmporium** [comment]
except that this time it didn't went down due to a DMCA, but a sort of Private Information issue. I've seen that links that could be viewed as "unauthorized" were put in that same package.

**Comment by BellezaEmporium** [comment]
https://gitlab.com/theofficialomega/backup in case you need.

**Reply by rizkidewanto** [reply]
The releases aren't backuped, same as link before

**Reply by BellezaEmporium** [reply]
> The releases aren't backuped, same as link before

Redoing all the pipelines and stuff isn't an affair of minutes, and I'm alone dealing with it for now. Due to lack of time I've only decided to upload the files as-is and added the built indexes.

**Comment by freearhey** [comment]
The discussion has been moved to _iptv-org/iptv-org.github.io_ repository due to GitHub's decision to disable "Team Discussions": https://github.blog/changelog/2023-02-08-sunset-notice-team-discussions/

**Comment by themak1985** [comment]
Wondering if this is related:
https://www.theverge.com/2023/2/28/23617624/pluto-tv-dmca-mpa-github-playlists

Also, not a developer, but happy to help in any way.

**Comment by Daniel15** [comment]
@iptv-org/core I just got a reply from GitHub, and it sounds like the repos were actually disabled due to the GitHub actions that were being used.

I can ask them if they can re-enable the repos but remove the Actions? We could just use a different CI system instead. Their message seems pretty final though 😕 

> Hi Daniel,
> 
> Thanks for reaching out.
> 
> Please accept our apologies, we sent an incorrect response for the actioning of your repositories that indicated they were disabled due to a violation of our Private Information Removal policy.
> 
> Your repositories were disabled following reports of conduct in violation of the following prohibition found in the GitHub Actions section of our [additional product terms](https://docs.github.com/site-policy/github-terms/github-terms-for-additional-products-and-features#actions).
> 
>  > Additionally, regardless of whether an Action is using self-hosted runners, Actions should not be used for:
> >
> >  any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden); or
> 
> Because of the nature and volume of the prohibited activity, we will not be reinstating your repositories at this time.
> 
> If you’d like to dispute that a violation occurred, please provide additional information to show that a different decision should have been reached.
> 
> We appreciate your understanding and apologize again for the miscommunication. Please let us know if you have any questions or concerns about this notice.
> 
> Regards,  
> GitHub Trust & Safety

**Reply by BellezaEmporium** [reply]
> @iptv-org/core I just got a reply from GitHub, and it sounds like the repos were actually disabled due to the GitHub actions that were being used.
> 
> I can ask them if they can re-enable the repos but remove the Actions? We could just use a different CI system instead. Their message seems pretty final though 😕 
> 
> > Hi Daniel,
> > 
> > Thanks for reaching out.
> > 
> > Please accept our apologies, we sent an incorrect response for the actioning of your repositories that indicated they were disabled due to a violation of our Private Information Removal policy.
> > 
> > Your repositories were disabled following reports of conduct in violation of the following prohibition found in the GitHub Actions section of our [additional product terms](https://docs.github.com/site-policy/github-terms/github-terms-for-additional-products-and-features#actions).
> > 
> >  > Additionally, regardless of whether an Action is using self-hosted runners, Actions should not be used for:
> > >
> > >  any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden); or
> > 
> > Because of the nature and volume of the prohibited activity, we will not be reinstating your repositories at this time.
> > 
> > If you’d like to dispute that a violation occurred, please provide additional information to show that a different decision should have been reached.
> > 
> > We appreciate your understanding and apologize again for the miscommunication. Please let us know if you have any questions or concerns about this notice.
> > 
> > Regards,  
> > GitHub Trust & Safety
> 
> 

Actually they acknowledge that they were faulty in the message.

I suppose we should do that. Finding CIs are quite easy nowadays.

**Reply by Daniel15** [reply]
Replied with this. Let's see what they say:

> So you just disable repositories with no warning? I didn't see any warnings about this at all. It seems like a more appropriate reaction would have been to just disable the actions instead.
> 
> Is it possible to reactivate the repos but disable the actions that violate these terms (or disable actions for the whole repo if you feel like that's the right thing to do)? We'd really like to avoid having to move these repos to a competing service.
> 
> Thanks,
> Daniel
>

**Reply by boeder9** [reply]
> @iptv-org/core I just got a reply from GitHub, and it sounds like the repos were actually disabled due to the GitHub actions that were being used.
> 
> I can ask them if they can re-enable the repos but remove the Actions? We could just use a different CI system instead. Their message seems pretty final though 😕
> 
> > Hi Daniel,
> > Thanks for reaching out.
> > Please accept our apologies, we sent an incorrect response for the actioning of your repositories that indicated they were disabled due to a violation of our Private Information Removal policy.
> > Your repositories were disabled following reports of conduct in violation of the following prohibition found in the GitHub Actions section of our [additional product terms](https://docs.github.com/site-policy/github-terms/github-terms-for-additional-products-and-features#actions).
> > > Additionally, regardless of whether an Action is using self-hosted runners, Actions should not be used for:
> > > any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden); or
> > 
> > 
> > Because of the nature and volume of the prohibited activity, we will not be reinstating your repositories at this time.
> > If you’d like to dispute that a violation occurred, please provide additional information to show that a different decision should have been reached.
> > We appreciate your understanding and apologize again for the miscommunication. Please let us know if you have any questions or concerns about this notice.
> > Regards,
> > GitHub Trust & Safety

Hmmm what "actions" are they referring to? I don't follow. This applies to both the streams and epg repos?

**Reply by Daniel15** [reply]
@boeder9  It's referring to the "GitHub Actions" product, which is GitHub's system that lets you run build scripts directly on GitHub.

**Reply by Alstruit** [reply]
I did remember there was a "certain" branch that probably overloaded the Github Actions. It could be linked to the main and the epg repo. The update frequency action should not have taken down the main and epg by itself.

**Reply by RevGear** [reply]
> @iptv-org/core I just got a reply from GitHub, and it sounds like the repos were actually disabled due to the GitHub actions that were being used.
> 
> > Your repositories were disabled following reports of conduct in violation of the following prohibition found in the GitHub Actions section of our [additional product terms](https://docs.github.com/site-policy/github-terms/github-terms-for-additional-products-and-features#actions).
> > > Additionally, regardless of whether an Action is using self-hosted runners, Actions should not be used for:
> > > any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden); or

So it was the amount of resources being used. Would be nice of it they would issue a warning first and allow people to remove/re-work some of the actions to decrease the load rather than just nuke the entire thing.

**Comment by misar1** [comment]
Being cynically inclined I wonder if the first justification was **not** a mistake and the situation was similar to the article linked by @themak1985   https://www.theverge.com/2023/2/28/23617624/pluto-tv-dmca-mpa-github-playlists
We now have an alternative justification which is directly supported by the T&Cs.

In discussing this it is important to distinguish between what violates legal rights (nothing on here did) and what content providers don't like. Above all they don't like any device or app which allows their content to be viewed outside their control. They have good reasons **from their point of view** when the content is unencrypted. Enforcing use of their apps gives them benefits such as control of quality and distribution of recordings or requiring registration so they know precisely who is watching what for advertising purposes. IPTV-ORG does not provide a device or app but it greatly facilitates using TV head ends, VLC, etc.

**Comment by Daniel15** [comment]
@iptv-org/core I convinced GitHub to restore the repositories, however we are no longer allowed to use Actions in these repos. We can replace it with an external CI service like CircleCI, or self-hosted Jenkins.

**Reply by nabeelr** [reply]
was that what they were upset by?
